client/src/legacy/GridField.js

```diff
@@ -180,6 +194,21 @@ $.entwine('ss', function($) {
       if (hasLegacyFilterHeader) {
         this.find('.sortable-header th:last').html(content);
       }
+    },
+
+    keepStateInHistory: function() {
+      const newURLSceme = $(this).find('.gridfield-actionmenu__container').data('schema');
+      if (newURLSceme && newURLSceme.length > 0) {
+        newURLSceme.filter( e => {
+          if (e.type === 'link') {
+            const url = [window.location.origin, e.url].join('/');
+            const params = (new URL(url)).searchParams.toString();
```

Comment:
This is still not retaining any GET params in the URL before sorting.
So if you start off at `http://my-local.dev/admin/gridfield/?some-var=some-val` the `some-var=some-val` won't be kept.
We need to make sure we keep the `window.location.search` params like so:
```suggestion
            const url = [window.location.origin, e.url].join('/');
            const params = [window.location.search, url.searchParams.toString()].join('&').substring(1);;
```
Note that if `window.location.search` has a value, it will include a `?` at the start, but if it _doesn't_ have a value, the result of the `join` will have a `&` at the start. So either way we need to trim off the first character of `params` which is what the `substring` is there for.

Reply:
I add params after request to URL (history.state) only to provide ability to share this result (link) and navigate by using Browser's buttons.

Reply:
We cannot keep custom params in URL from POST request.

Reply:
> to provide ability to share this result (link) and navigate by using Browser's buttons.

Yes, that's exactly my point. When navigating back/forward using the browser's back/forward buttons, we should keep any custom GET variables in the query string that have been added through project code, as it may be important for custom functionality.

> We cannot keep custom params in URL from POST request.

That's absolutely fine. The recommended change _won't_ keep custom POST params in the URL, nor would we ever want to.